Adding JFrog Xray Violations and Logs Integrations (#883)
DataDog/integrations-extras: jfrog_platform/README.md

## Setup

### Requirements
* Your [Datadog API key][6].

### Logs Collection

1. Fluentd installation :

    Please follow the detailed steps for Fluentd instllation based on your installation type and defining the environment variable correctly [here][18]  
 
2. Fluentd configuration : 
        

>     #### Artifactory Configuration

Reviewer: ```suggestion
   **Artifactory Configuration**
```
using a h4 in an ordered list won't render, so I suggest updating it to bold

Reviewer: It also might be good to break this section into a, b, c sub steps

>     Download the artifactory fluentd configuration file to a directory the user has permissions to write, such as the $JF_PRODUCT_DATA_INTERNAL locations discussed above in the [Environment Configuration](#environment-configuration) section.

Reviewer: ```suggestion
    Download the artifactory fluentd configuration file to a directory the user has permissions to write, such as the `$JF_PRODUCT_DATA_INTERNAL` locations discussed in the [Environment Configuration](#environment-configuration) section.
```

    
    ````text
    cd $JF_PRODUCT_DATA_INTERNAL
    wget https://raw.githubusercontent.com/jfrog/log-analytics-datadog/master/fluent.conf.rt
    ````
    

>     Override the match directive(last section) of the downloaded `fluent.conf.rt` with the details given below

Reviewer: ```suggestion
    Override the match directive (last section) of the downloaded `fluent.conf.rt` with the details given below:
```

    
    ```
    <match jfrog.**>
      @type datadog
      @id datadog_agent_jfrog_artifactory
      api_key API_KEY
      include_tag_key true
      dd_source fluentd
    </match>
    ```
    
    _**required**_: ```API_KEY``` is the apiKey from [Datadog][4]
    
    ```dd_source``` attribute is set to the name of the log integration in your logs in order to trigger the integration automatic setup in datadog.
    
    ```include_tag_key``` defaults to false and it will add fluentd tag in the json record if set to true
    
    #### Xray Configuration
    
    Download the Xray fluentd configuration file to a directory the user has permissions to write, such as the $JF_PRODUCT_DATA_INTERNAL locations discussed above in the [Environment Configuration](#environment-configuration) section.
    
    ````text
    cd $JF_PRODUCT_DATA_INTERNAL
    wget https://raw.githubusercontent.com/jfrog/log-analytics-datadog/master/fluent.conf.xray
    ````
    

>     Fill in the JPD_URL, USER, JFROG_API_KEY fields in the source directive of the downloaded `fluent.conf.xray` with the details given below

Reviewer: ```suggestion
    Fill in the `JPD_URL`, `USER`, `JFROG_API_KEY` fields in the source directive of the downloaded `fluent.conf.xray` with the details given below:
```

    </source>
    ```
    
    _**required**_: ```JPD_URL``` is the Artifactory JPD URL of the format `http://<ip_address>` with is used to pull Xray Violations
    
    _**required**_: ```USER``` is the Artifactory username for authentication
    
    _**required**_: ```JFROG_API_KEY``` is the [Artifactory API Key][19] for authentication
    
    Override the match directive (last section) of the downloaded `fluent.conf.xray` with the details given below
    
    ```
    <match jfrog.**>
      @type datadog
      @id datadog_agent_jfrog_xray
      api_key API_KEY
      include_tag_key true
      dd_source fluentd
    </match>
    ```
    
    _**required**_: ```API_KEY``` is the apiKey from [Datadog][4]
    
    ```dd_source``` attribute is set to the name of the log integration in your logs in order to trigger the integration automatic setup in datadog.
    
    ```include_tag_key``` defaults to false and it will add fluentd tag in the json record if set to true
    
3. Integration enablement

    To enable this integration, run the td-agent on `artifactory`\ `xray` instances:

    ``` 
    td-agent
    ```

>     The API key is configured in `td-agent`, which will start sending logs to Datadog. For other types of installation, please refer to the [original documentation][18]

Reviewer: ```suggestion
    The API key is configured in `td-agent`, which will start sending logs to Datadog. For other types of installation, see the [jFrog documentation][18]
```
